### PR TITLE
adding property methods  to ivy Shape Class

### PR DESCRIPTION
**Pull request by soma2000-lang**
In continuation from #14760

Hi @Aarsh2001,one doubt,I could not find some reference for numel and count.
image

### CODE REVIEW

File: ivy/__init__.py [hunk 1]
```diff
@@ -355,6 +355,120 @@ def __dir__(self):
...
+    def index(self, index):
+        assert isinstance(self._shape, Shape)
+        if self._shape.rank is None:
+            return Shape(None)
+        else:
+            return self._shape[index]
+
+    @property
+    def shape(self):
+        return self._shap
```
Comment (Aarsh2001) [inline ivy/__init__.py:377]: Hey @soma2000-lang i think the shape property already exists above, you can remove this

**Review by Aarsh2001 — CHANGES_REQUESTED** [review]

File: ivy/__init__.py [hunk 2]
```diff
@@ -355,6 +355,120 @@ def __dir__(self):
...
+
+    @property
+    def assert_is_fully_defined(self):
+        if not self.is_fully_defined():
+            raise ValueError("Shape %s is not fully defined" % self)
+
+    @property
+    def as_list(self):
+        if self._dims is None:
+            raise ValueError("as_list() is not defined on an unknown TensorShape.")
```
Comment (Aarsh2001) [inline ivy/__init__.py:469]: It shouldn't be TensorShape here

**Review by Aarsh2001 — CHANGES_REQUESTED** [review]

**Comment by soma2000-lang** [comment]
Hi, @Aarsh2001 ,I will write the tests.

**Review by VedPatwardhan — COMMENTED** [review]
Hey @soma2000-lang and @Aarsh2001, feel free to re-request a review once the tests are added, I think the changes suggested by @Aarsh2001 are sufficient for now. Thanks for looking into this @soma2000-lang 🙂

**Review by vedpatwardhan — APPROVED** [review]
Hey @soma2000-lang, thanks for writing the tests, the changes look good to me, there's a few lint failures which the gardener would be able to fix, feel free to merge the PR once @Aarsh2001 is done reviewing 🙂

**Comment by vedpatwardhan** [comment]
ivy-gardener

**Comment by soma2000-lang** [comment]
ivy-gardener
white_check_mark Ivy gardener has formatted your code.
If changes are requested, don't forget to pull your fork.

**Review by Aarsh2001 — APPROVED** [review]
LGTM !

**Pull request merged**